pkg/providers/amifamily/bootstrap/bottlerocketsettings.go

```diff
@@ -43,43 +43,54 @@ type BottlerocketSettings struct {
+	EvictionHard                       map[string]string                         `toml:"eviction-hard,omitempty"`
+	KubeReserved                       map[string]string                         `toml:"kube-reserved,omitempty"`
+	SystemReserved                     map[string]string                         `toml:"system-reserved,omitempty"`
+	AllowedUnsafeSysctls               []string                                  `toml:"allowed-unsafe-sysctls,omitempty"`
+	ServerTLSBootstrap                 *bool                                     `toml:"server-tls-bootstrap,omitempty"`
+	RegistryQPS                        *int                                      `toml:"registry-qps,omitempty"`
+	RegistryBurst                      *int                                      `toml:"registry-burst,omitempty"`
+	EventQPS                           *int                                      `toml:"event-qps,omitempty"`
+	EventBurst                         *int                                      `toml:"event-burst,omitempty"`
+	KubeAPIQPS                         *int                                      `toml:"kube-api-qps,omitempty"`
+	KubeAPIBurst                       *int                                      `toml:"kube-api-burst,omitempty"`
+	ContainerLogMaxSize                *string                                   `toml:"container-log-max-size,omitempty"`
+	ContainerLogMaxFiles               *int                                      `toml:"container-log-max-files,omitempty"`
+	CPUManagerPolicy                   *string                                   `toml:"cpu-manager-policy,omitempty"`
+	CPUManagerReconcilePeriod          *string                                   `toml:"cpu-manager-reconcile-period,omitempty"`
+	TopologyManagerScope               *string                                   `toml:"topology-manager-scope,omitempty"`
+	ImageGCHighThresholdPercent        *string                                   `toml:"image-gc-high-threshold-percent,omitempty"`
+	ImageGCLowThresholdPercent         *string                                   `toml:"image-gc-low-threshold-percent,omitempty"`
+	CPUCFSQuota                        *bool                                     `toml:"cpu-cfs-quota-enforced,omitempty"`
+	ShutdownGracePeriod                *string                                   `toml:"shutdown-grace-period,omitempty"`
+	ShutdownGracePeriodForCriticalPods *string                                   `toml:"shutdown-grace-period-for-critical-pods,omitempty"`
 }
 
 type BottlerocketStaticPod struct {
 	Enabled  *bool   `toml:"enabled,omitempty"`
 	Manifest *string `toml:"manifest,omitempty"`
 }
 
+// BottlerocketCredentialProvider is k8s specific configuration for Bottlerocket Kubelet image credential provider
+// https://bottlerocket.dev/en/os/1.18.x/api/settings/kubernetes/#credential-providers
+// https://kubernetes.io/docs/tasks/administer-cluster/kubelet-credential-provider/
+type BottlerocketCredentialProvider struct {
```

Comment:
Can we also reference the GH link to this schema? e.g. https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/modeled-types/src/kubernetes.rs#L1307

Reply:
Done